messagecatalog/spec.md

```diff
@@ -0,0 +1,1475 @@
+            },
+            "type" : {
+              "value" : "io.cloudevents.messagecatalog.deleted",
+              "required" : true,
+              "description" : "type of this event"
+            }
+          }
+        }
+    },
+    {
+        "version" : 2,
+        "definition" : {
+          "attributes" : {
+            "specversion" : {
+              "value" :"1.0",
+              "required" : true,
+              "description" : "CloudEvents specification version"
+            },
+            "type" : {
+              "value" : "io.cloudevents.messagecatalog.deleted",
+              "required" : true,
+              "description" : "type of this event"
+            },
+            "subject" : {
+              "required" : true,
+              "description" : "id of the deleted event definition"
+            }
+          }
+        }
+    }
+  ]
+}
+```
+
+For version "2" of the above definition, this is the standalone _message
+definition version_ document:
+
+``` JSON
+{
+  "id" : "io.cloudevents.messagecatalog.deleted",
+  "authority" : "cloudevents.io",
+  "description" : "Raised when an object has been deleted from the catalog",
+  "format" : "CloudEvents",
+  "version" : 2,
+  "definition" : {
+    "attributes" : {
+      "specversion" : {
+        "value" :"1.0",
+        "required" : true,
+        "description" : "CloudEvents specification version"
+      },
+      "type" : {
+        "value" : "io.cloudevents.messagecatalog.deleted",
+        "required" : true,
+        "description" : "type of this event"
+      },
+      "subject" : {
+        "required" : true,
+        "description" : "id of the deleted event definition"
+      }
+    }
+  }
+}
+```
+
+The API is created such that a message definition MUST be created by posting an
+initial version and that the message definition (including any common parts like
+the documentation) MUST only be changed by posting a new version. Existing
+versions can be individually retrieved and deleted, but not modified.
+
+A newer message definition version might introduce breaking changes or it might
+only introduce careful changes that preserve compatibility. These strategies are
+not subject of this specification, but the API provides a conflict handling
+mechanism that allows an implementation to reject updates that do not comply
+with a compatibility policy, if one has been implemented.
+
+For simple scenarios, the API allows for version management to be automatic and
+transparent. Whenever a message definition is updated, a new version number is
+assigned and prior definition versions are retained. This specification does not
+mandate a retention policy, but implementations MAY retire and remove outdated
+definition versions.
+
+The latest available definition MUST always the default version that is
+retrieved when the URI to the Message definition is used without the version
+having been specified.
+
+### 2.2.1 Authority
```

Comment:
I suggest adding audience as well. Then you have symmetry on `who` and `for whom`